### PR TITLE
[Artwork] Add tracking to OtherWorks ArtworkGrid bricks

### PR DESCRIPTION
**Pull request by damassi**
This updates the OtherWorks grids with new tracking functionality, but two key parts had to be extended:
- `ArtworkGrid` now accepts an optional `onBrickClick` callback prop
- `GridItem` now accepts an optional `onClick` callback prop

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 8d80107ce886206710fd5b9d977598a5d878dc17

https://deploy-preview-1816--artsy-reaction.netlify.com

File: src/Apps/Artwork/Components/OtherWorks/ArtworkContexts/ArtworkGrids/ArtistArtworkGrid.tsx [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
+}
+
+@track({
+  context_module: Schema.ContextModule.OtherWorksByArtist,
+})
+export class ArtistArtworkGrid extends React.Component<ArtistArtworkGridProps> {
+  @track({
+    type: Schema.Type.ArtworkBrick,
+  })
+  trackBrickClick(p) {
```
Comment (mzikherman) [inline src/Apps/Artwork/Components/OtherWorks/ArtworkContexts/ArtworkGrids/ArtistArtworkGrid.tsx]: what's p?

File: src/Apps/Artwork/Components/OtherWorks/ArtworkContexts/ArtworkGrids/ArtistArtworkGrid.tsx [hunk 2]
```diff
@@ -24,10 +45,15 @@ export const ArtistArtworkGridFragmentContainer = createFragmentContainer<{
...
+          onBrickClick={this.trackBrickClick.bind(this)}
         />
       </>
     )
-  }),
+  }
+}
+
+export const ArtistArtworkGridFragmentContainer = createFragmentContainer(
+  withContext(ArtistArtworkGrid),
```
Comment (mzikherman) [inline src/Apps/Artwork/Components/OtherWorks/ArtworkContexts/ArtworkGrids/ArtistArtworkGrid.tsx]: oo this is a nicer way than the more verbose + indentation with `<ContextConsumer...` style.

**Comment by mzikherman** [comment]
Merge when ready!

File: src/Apps/Artwork/Components/OtherWorks/ArtworkContexts/ArtworkGrids/ArtistArtworkGrid.tsx (hunk 2 above)
Comment (damassi) [reply src/Apps/Artwork/Components/OtherWorks/ArtworkContexts/ArtworkGrids/ArtistArtworkGrid.tsx]: Yah agree, tis was a nice experiment, but this is more flexible

**Pull request merged**